python/lsst/meas/algorithms/sourceSelector.py

```diff
@@ -102,19 +106,23 @@ def run(self, sourceCat, sourceSelectedField=None, matches=None, exposure=None):
         RuntimeError
             Raised if ``sourceCat`` is not contiguous.
         """
-        if not sourceCat.isContiguous():
-            raise RuntimeError("Input catalogs for source selection must be contiguous.")
+        if not isinstance(sourceCat, DataFrame) and not isinstance(sourceCat, Table):
```

Comment:
I think you want this to be `if isinstance(sourceCat, lsst.afw.table.SimpleCatalog)`, since that is explicitly the one that needs a contiguity check.